core/server/worker/src/main/java/alluxio/worker/page/PagedBlockStoreMeta.java

```diff
@@ -0,0 +1,198 @@
+import alluxio.client.file.cache.PageId;
+import alluxio.client.file.cache.PageInfo;
+import alluxio.client.file.cache.PageMetaStore;
+import alluxio.client.file.cache.store.PageStoreDir;
+import alluxio.collections.Pair;
+import alluxio.exception.PageNotFoundException;
+import alluxio.resource.LockResource;
+import alluxio.worker.block.BlockStoreLocation;
+import alluxio.worker.block.BlockStoreMeta;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Multimaps;
+import com.google.common.collect.Streams;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+/**
+ * Metadata of the paged block store.
+ */
+public class PagedBlockStoreMeta implements BlockStoreMeta {
+  public static final String DEFAULT_TIER = "DefaultTier";
+  public static final String DEFAULT_DIR = "DefaultDir";
+  public static final String DEFAULT_MEDIUM = "SSD";
+  public static final DefaultStorageTierAssoc DEFAULT_STORAGE_TIER_ASSOC =
+      new DefaultStorageTierAssoc(ImmutableList.of(DEFAULT_MEDIUM));
+
+  private final long mCapacity;
+  private final long mUsed;
+  private final int mNumBlocks;
+  private final List<String> mDirPaths;
+  private final List<Long> mCapacityBytesOnDirs;
+  private final List<Long> mUsedBytesOnDirs;
+  @Nullable
+  private final Map<BlockStoreLocation, List<Long>> mBlockLocations;
+  @Nullable
+  private final List<Long> mBlocks;
+
+  /**
+   * Creates a new snapshot of the state of the paged block store.
+   * @param metaStore the meta store
+   * @param shouldIncludeBlockIds whether detailed block info should be included in the snapshot. If
+   *                              false, {@link #getBlockList} and
+   *                              {@link #getBlockListByStorageLocation()} will return {@code null}.
+   */
+  PagedBlockStoreMeta(PageMetaStore metaStore, boolean shouldIncludeBlockIds) {
+    // storage dir list is immutable, so no need to acquire a read lock here
+    final List<PageStoreDir> dirs = metaStore.getStoreDirs();
+    mDirPaths =
+        dirs.stream().map(dir -> dir.getRootPath().toString()).collect(Collectors.toList());
+    mCapacityBytesOnDirs =
+        dirs.stream().map(PageStoreDir::getCapacityBytes).collect(Collectors.toList());
+    mUsedBytesOnDirs =
+        dirs.stream().map(PageStoreDir::getCachedBytes).collect(Collectors.toList());
+    mCapacity = mCapacityBytesOnDirs.stream().reduce(0L, Long::sum);
+    mUsed = mUsedBytesOnDirs.stream().reduce(0L, Long::sum);
+
+    if (shouldIncludeBlockIds) {
+      HashMultimap<PageStoreDir, String> dirToBlocksMap = HashMultimap.create();
+      try (LockResource lock = new LockResource(metaStore.getLock().readLock())) {
+        final Iterator<PageId> pages = metaStore.getPagesIterator();
```

Comment:
PageStoreDir maintains a set of file ids, so actually we don't need to iterate the pages.  But let's keep the impl for now then we could start the tests earlier, because the file_ids in the dir is not fully ready.

Reply:
Added a todo.